src/services/revocation.js

```diff
@@ -32,60 +32,183 @@ export class RevocationStatusServiceImpl {
    * Checks revocation status of UCAN delegations via Storage UCAN Service
    *
    * @param {Ucanto.Proof[]} proofs - Array of UCAN proofs to check
+   * @param {string} spaceDID - Space DID to validate delegation context
    * @param {import('../types/env.js').Env} env - Environment configuration
    * @returns {Promise<import('@ucanto/server').Result<boolean, import('@ucanto/server').Failure>>}
    */
-  async checkStatus (proofs, env) {
-    const safeProofs = proofs || []
-
+  async checkStatus (proofs, spaceDID, env) {
+    const delegations = (proofs || []).map(p => /** @type {import('@ucanto/interface').Delegation} */ (p))
```

Comment:
You need to assert it is a delegation not a CID, not just type cast. Use the `isDelegation` function.

Reply:
Should probably fail if the proof is a CID, or take a proof resolver function as an option (like this: https://github.com/storacha/ucanto/blob/d7420d5f96962a05652dd165c1fd26fa1d140f47/packages/interface/src/lib.ts#L731).

Reply:
That said, if you use the authorization object passed to the `verifyAuthotization` callback, it'll have already resolved proofs or failed as necessary.